pkg/cgroup/cgroup.go

```diff
@@ -58,10 +64,41 @@ type Cgroups struct {
 }
 
 func NewCgroups() (*Cgroups, error) {
-	var err error
 	var cgrp *Cgroup
 	var cgroupv1, cgroupv2 Cgroup
 
+	// lock thread so that namespace switch will apply in the same thread
+	// we mount the cgroupfs.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// switch to root cgroup namespace before attempting initialization
+	err := capabilities.GetInstance().Specific(
+		func() error {
+			hostCgroupNsFile, err := os.Open("/proc/1/ns/cgroup")
+			if err != nil {
+				return fmt.Errorf("failed to move to host cgroupns: %v", err)
+			}
+			defer func() {
+				err := hostCgroupNsFile.Close()
+				if err != nil {
+					logger.Errorw("failed to close file /proc/1/ns/cgroup when moving to host cgroup ns: %v", err)
+				}
+			}()
+			var fd = int(hostCgroupNsFile.Fd())
+			err = unix.Setns(fd, unix.CLONE_NEWCGROUP)
```

Comment:
Is this thread returning to its previous cgroup ns somewhere? If not, wouldn't this logic be in a temp goroutine call? WDYT?

Reply:
No, supposedly golang can execute functions on different threads opportunistically. This ensures that the mounting process will occur in the same thread as the one we changed cgroupns in.
I've actually kept this code so that tracee is always deployed in the root cgroupns. Might be more appropriate to do in a cgo constructor when tracee executes, since that is done in a single threaded context and as such will propagate to tracee's threads as a whole.

Then again, removing this code hasn't affected functionality of the fix as far as i've tested. So maybe we should drop the commit. WDYT?

Reply:
> Then again, removing this code hasn't affected functionality of the fix as far as i've tested. So maybe we should drop the commit. WDYT?

I wouldn't drop it, but I would insert a comment about the cgo constructor and/or that the current caller's goroutine will possibly be in a different cgroup mount ns than the batch. It would help us in the future.

Reply:
Done